src/inmanta/agent/handler.py

```diff
@@ -288,15 +313,37 @@ def action_id(self) -> uuid.UUID:
     def status(self) -> Optional[const.ResourceState]:
         return self._status
 
+    @property
+    def resource_state(self) -> Optional[ContextResourceState]:
+        return self._resource_state
+
     @property
     def logs(self) -> list[data.LogLine]:
         return self._logs
 
     def set_status(self, status: const.ResourceState) -> None:
         """
-        Set the status of the handler operation.
+        Set the status of the handler operation and translate it to ContextResourceState
         """
         self._status = status
+        try:
+            self._resource_state = ContextResourceState(status)
+        except ValueError:
+            # How to proceed? What about unavailable?
+            self.logger.warning(
+                "Called set_status with status %s which is not directly translatable to NewResourceState", status
+            )
```

Comment:
1. This will become a failure, which is fine I think
2. I don't think unavailable can/should ever be set here, as it indicates we can't load the handler, so no ctx is constructed I think?

Reply:
I'm starting to think that we need the unavailable state.
Here we can't find the handler but we have a context built: https://github.com/inmanta/inmanta-core/pull/8383/files#diff-f34628194310d50c9be6ca572bc1fccf6323eb8c0bf735df244e74b91e3e9b7eL163

And it is useful for when we call `DeployResult.undeployable` (assuming we want the state there to be `unavailable` instead of `failed`)

What do you think?